### ISSUE TITLE
Add `frolvlad/alpine-glibc` as a base image

### ISSUE THREAD

**Issue by wholtz** [mamba-org/micromamba-docker#236]
See discussion in #235 . We should be able to add `frolvlad/alpine-glibc` as a base image if we install bash as part of the build process.

- [x] Update Dockerfile to handle both debian-based and alpine-based base-images (package installation methods, SSL file paths)
- [x] Update shell scripts to be compatible with alpine-based base-images containing bash (group/user add commands) 
- [x] Update tests to be compatible with alpine user/group administration commands

**Comment by wholtz** [comment]
I need a way to update a username in alpine linux. The best I have so far is using sed:

```
sed 's%mambauser:\([^:]*\):\(\d*\):\(\d*\):\([^:]*\):\([^:]*\):%new_user:\1:\2:\3:\4:/home/new_user:%' < 
/etc/passwd
```

**Comment by maresb** [comment]
That looks pretty effective, though not easy on the eyes. But without `usermod` I don't see much choice.

There might be some edge cases missing. Like there should probably be a beginning-of-line `^` anchor (and perhaps the EOL `$` just for good measure, though this may lead to problems...). Also I'm not sure if escaped colons might occur in the directory name. There is also the optional GECOS field, so some lines can have an additional `:`-separated entry.

I think you could bypass the above complications with the pattern `^mambauser:\([^:]*\):\(\d*\):\(\d*\):`.

Alternatively, you could add the `usermod` command as per https://github.com/chrootlogin/docker-nextcloud/issues/3#issuecomment-271626117, although for that I'd recommend `http`→`https` and somehow removing the extra line in `/etc/apk/repositories` afterwards.

**Comment by wholtz** [comment]
This is working and ready to go on the [`alpine` branch](https://github.com/mamba-org/micromamba-docker/tree/alpine). I'm going to hold off merging this into main until the next minor version release of micromamba occurs.

**Comment by maresb** [comment]
Wow, thanks so much for getting this running!!!

The one thing that caught my eye is that it looks like alpine has no `/etc/skel`, so you added the init script to `/etc/profile.d`. I can never remember for which combos of login/interactive shells Bash runs `/etc/profile.d`, but it just made me wonder if we would break anything by using that instead of `~/.bashrc`. The advantage would be more uniformity between Alpine and Ubuntu.

**Comment by wholtz** [comment]
Thanks for the feedback @maresb. I did a little more looking, and I can use `/etc/skel` on alpine, I just need to make the directory first. Additionally, I tested using `/etc/profile.d` on debian instead of `/etc/skel` and it resulted in tests failure (but I didn't do any investigation). 

This also made me notice a corner case that currently isn't being tested..... so I'm probably going to do a bit more work to get this ready for merging.
